doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/service-query-con.adoc

```diff
@@ -0,0 +1,126 @@
+[id='service-query-con_{context}']
+= Query service
+
+The query service provides advanced search capabilities that are based on Dashbuilder DataSets.
+
+With this approach, you can control how to retrieve data from underlying data store. You can use complex JOIN statements with external tables such as JPA entities tables, custom systems database tables, and so on.
+
+The query service is build around the following two sets of operations:
+
+* Management operations:
+** Register query definition
+** Replace query definition
+** Unregister (remove) query definition
+** Get query definition
+** Get all registered query definitions
+* Runtime operations: 
+** simple query based on `QueryParam` as filter provider
+** advanced query based on `QueryParamBuilder` as filter provider
+
+DashBuilder DataSets provide support for multiple data sources, such as CSV, SQL, and Elastic Search. However, the {PROCESS_ENGINE} uses a RDBMS based backend and focuses on SQL-based data sets.
+
+Therefore, the {PROCESS_ENGINE} query service is a subset of DashBuilder DataSets capabilities that enables efficient queries with a simple API.
+
+[float]
+== Key classes of the query service
```

Comment:
Because this module file is already level offset of 3, these subsections are pushed to level 4 and level 5 offset, which makes it render very poorly. We try to stick to 3 levels of offset max, with exceptions for 4. I don't think we want 5 ever. However, this is a genuine problem in complex documents like this with many aspects of many pieces.

That said, to avoid excessive offset, we can just avoid unnecessary titles when practical. In this case, I think the titles are unnecessary anyway and so an easy fix, imo.

I'd just nix the titles and run the text as follows:

```
... queries with a simple API.

The query service uses the following key classes:
...

The `QueryResultMapper` class maps ...
...

The `QueryParamBuilder` class provides ...
...
```

(Also note that the "Key classes" title would've needed a lead-in sentence anyway.)